### PR TITLE
[BE] Enforce missing `override` keyword

### PR DESCRIPTION
**Pull request by cyyever**
This PR enables `-Winconsistent-missing-destructor-override` and `-Winconsistent-missing-override`
and fixes violations.

### <samp>🤖 Generated by Copilot at 47e904e</samp>

This pull request updates the code of various classes and operators in the `caffe2` and `aten` subdirectories to use the `override` specifier instead of the `virtual` keyword for destructors and other virtual functions that override a base class function. This improves the code readability, quality, and consistency with C++ best practices. It also modifies the `./CMakeLists.txt` file to enable warnings for these specifiers, but disable errors.

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -817,6 +817,8 @@ if(NOT MSVC)
   append_cxx_flag_if_supported("-Wno-strict-aliasing" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Wvla-extension" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Wnewline-eof" CMAKE_CXX_FLAGS)
+  append_cxx_flag_if_supported("-Winconsistent-missing-override" CMAKE_CXX_FLAGS)
+  append_cxx_flag_if_supported("-Winconsistent-missing-destructor-override" CMAKE_CXX_FLAGS)
```
Comment (malfet) [inline CMakeLists.txt:821]: Just curious: aren't those part of `-Wall`?

Comment (cyyever) [reply CMakeLists.txt:821]: No, as you can see from clang doc

Comment (malfet) [reply CMakeLists.txt:821]: If you ever plan to do the followup PR, mind adding a comments explaining why those  must be added on top of `-Wall` and `-Wextra` flags (as I've assumed those two should be all inclusive)

**Pull request closed without merging**